Libraries/rocSOLVER/syev_batched/main.cpp

```diff
@@ -0,0 +1,264 @@
+    std::vector<rocblas_double> W(batch_eigenvalue_num_elements); // Resulting eigenvalues.
+    std::vector<rocblas_int>    info(batch_size); // Convergence information.
+
+    // 3. Generate a random symmetric matrix.
+    std::default_random_engine                     generator;
+    std::uniform_real_distribution<rocblas_double> distribution(0., 2.);
+    auto random_number = [&]() { return distribution(generator); };
+
+    for(int k = 0; k < batch_size; ++k)
+    {
+        int offset = k * single_matrix_num_elements;
+        for(int i = 0; i < n; ++i)
+        {
+            A[(n + 1) * i + offset] = random_number();
+            for(int j = 0; j < i; ++j)
+            {
+                A[i * n + j + offset] = A[j * n + i + offset] = random_number();
+            }
+        }
+    }
+
+    // 4. Set rocSOLVER parameters.
+    const rocblas_evect evect = rocblas_evect::rocblas_evect_original;
+    const rocblas_fill  uplo  = rocblas_fill::rocblas_fill_lower;
+
+    // 5. Reserve and copy data to device.
+    rocblas_double*  array_A[batch_size];
+    rocblas_double** array_dA;
+    rocblas_double*  d_A    = nullptr;
+    rocblas_double*  d_W    = nullptr;
+    rocblas_int*     d_info = nullptr;
+
+    const rocblas_int info_size  = sizeof(rocblas_int) * batch_size;
+    const rocblas_int array_size = batch_size * sizeof(rocblas_double**);
+
+    const rocblas_int full_size = batch_size * single_matrix_num_elements * sizeof(rocblas_double*);
+    const rocblas_int batch_eigenvalue_size = sizeof(double) * batch_eigenvalue_num_elements;
+
+    HIP_CHECK(hipMalloc(&d_info, info_size));
+    HIP_CHECK(hipMalloc(&d_A, full_size));
+    HIP_CHECK(hipMalloc(&d_W, batch_eigenvalue_size));
+    HIP_CHECK(hipMalloc(&array_dA, array_size));
+
+    HIP_CHECK(hipMemcpy(d_A, A.data(), full_size, hipMemcpyHostToDevice));
+    for(int k = 0; k < batch_size; ++k)
+    {
+        array_A[k] = &d_A[k * single_matrix_num_elements];
+    }
+    HIP_CHECK(hipMemcpy(array_dA, array_A, array_size, hipMemcpyHostToDevice));
+
+    // 6. Initialize rocBLAS.
+    rocblas_handle handle;
+    ROCBLAS_CHECK(rocblas_create_handle(&handle));
+
+    // 7. Get and reserve the working space on device.
+    const rocblas_int batch_tridiagonal_size
+        = batch_eigenvalue_num_elements * sizeof(rocblas_double);
+    rocblas_double* d_work = nullptr;
+    HIP_CHECK(hipMalloc(&d_work, batch_tridiagonal_size));
+
+    // 8. Compute eigenvectors and eigenvalues.
+    ROCBLAS_CHECK(rocsolver_dsyev_batched(handle,
+                                          evect,
+                                          uplo,
+                                          n,
+                                          array_dA,
+                                          lda,
+                                          d_W,
+                                          batch_size,
+                                          d_work,
+                                          n * n,
```

Comment:
This should be n.

Using `n*n` as the stride will probably segfault when n>1 and batch_size>1, since d_work only has n elements per batch and not `n*n`.

Reply:
Actually, I believe it should be `3 * n - 1` for the reasons already stated in previous comments. I defined
```
const rocblas_int single_tridiagonal_num_elements = n * 3 - 1;
const rocblas_int batch_tridiagonal_num_elements  = single_tridiagonal_num_elements * batch_size;
```
and changed the workspace size (currently it was `n * batch_size`) to `batch_tridiagonal_num_elements` and the mentioned parameter to `single_tridiagonal_num_elements`.

Also, the parameter `batch_size` passed right after `d_W` should be `n`, because it's supposed to be the stride between the eigenvalues vector for each matrix in the batch. I added a `single_eigenvalue_num_elements = n` constant and used that instead.